rust/core-lib/src/rpc.rs

```diff
@@ -43,6 +43,8 @@ pub enum Request<'a> {
 #[derive(Debug, PartialEq, Eq)]
 pub enum CoreCommand<'a> {
     Edit { view_id: &'a str, edit_command: EditCommand<'a> },
+    /// A command from the client to a plugin.
+    Plugin {  plugin_command: PluginCommand },
```

Comment:
yea, the way this is structured is the result of me wanting to use serde to handle generating the enums, and it not being easy to do as you described. Ideally `view_id` would be threaded that way as well.

I'm not sure if this is doable and I just haven't done it, or if there needs to be another approach.